src/cockpit/beiboot.py

```diff
@@ -130,10 +130,25 @@ class AuthorizeResponder(ferny.AskpassHandler):
     commands = ('ferny.askpass', 'cockpit.report-exists')
     router: Router
 
-    def __init__(self, router: Router):
+    def __init__(self, router: Router, basic_password: Optional[str]):
         self.router = router
+        self.basic_password = basic_password
+        self.have_basic_password = bool(basic_password)
 
     async def do_askpass(self, messages: str, prompt: str, hint: str) -> Optional[str]:
+        logger.debug("AuthorizeResponder: prompt %r, messages %r, hint %r", prompt, messages, hint)
+
+        if self.have_basic_password and 'password:' in prompt.lower():
+            # the UI currently expects us to fail on a wrong password, so only send it once
+            if self.basic_password is not None:
+                logger.debug("AuthorizeResponder: sending Basic auth password for prompt %r", prompt)
+                reply = self.basic_password
+                self.basic_password = None
+                return reply
+            else:
+                logger.debug("AuthorizeResponder: prompt %r, used up our password, failing", prompt)
+                return messages.strip().splitlines()[-1]
```

Comment:
What is this trying to do, exactly?

Reply:
It returns the last line of ssh (which is usually something like "wrong password blabla") to the UI via becoming the `message` bit of the generated CockpitError in the `except ferny.InteractionError` interpretation in `run()`. I previously raised an `authentication-failed` CockpitError directly, but you asked me to convert it to stderr.

Reply:
See previous comment, turns out we don't need this any more.